index.js

```diff
@@ -132,9 +132,10 @@ exports.getKeyIndexes = function (commandName, args, options) {
       }
       break
     case 'xreadgroup':
+    case 'xread':
       for (i = 0; i < args.length - 1; i++) {
-        if (args[i].toUpperCase() === 'STREAMS') {
-          for (var j = i + 1; j < (args.length - 1); j++) {
+        if (String(args[i]).toUpperCase() === 'STREAMS') {
+          for (j = i + 1; j <= i + ((args.length - 1 - i) / 2); j++) {
```

Comment:
Thanks for this!

nitpicky microoptimisation -- for `xread` you need to start at 0, but for `xreadgroup` you can skip the `GROUP group consumer` arguments. This almost certainly doesn't matter, but thought I'd mention it.

Reply:
Thank you, I was considering it, but it should be another `if` or another `case`... So i left it like it is :-).

Reply:
It could be implemented in line 136 as:

```js
      for (i = commandName === 'xred' ? 0 : 3; i < args.length - 1; i++) {
```